### PR TITLE
Gitian

### PR DESCRIPTION
**Pull request by josephbisch**
@droark 

This PR aims to allow reproducible building of Armory.app using Gitian.

Currently all the dependencies build using Gitian, but when Armory's configure script runs, it cannot find PyQt. The issue is that there are two Pythons (one for the host and one for the build system). The PyQt check tries to run the python interpreter and import PyQt to see if it can be done successfully. The issue is that we run the build Python, while PyQt is only available to the host Python, which we can't run on the build system. I think the best solution is to only run the check if the build system architecture is the same as that of the host system, if we run the check at all.

### CODE REVIEW

File: depends/packages/openssl.mk [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # Pretty much taken verbatim from Bitcoin Core
 package=openssl
-$(package)_version=1.0.1m
+$(package)_version=1.0.1k
```
Comment (droark) [inline depends/packages/openssl.mk]: Why the downgrade?

File: depends/patches/python3/0010-cross-darwin-feature.patch [hunk 2]
```diff
@@ -0,0 +1,120 @@
+diff -urN a-3.4.x/configure.ac b-3.4.x/configure.ac
```
Comment (droark) [inline depends/patches/python3/0010-cross-darwin-feature.patch]: Where did this patch come from?

**Comment by droark** [comment]
Thanks. Looks pretty good overall. The PyQt issue is a tricky one. Your solution is probably the best one overall. You may still want to ask about it on the [PyQt mailing list](http://www.riverbankcomputing.com/mailman/listinfo/pyqt) for a second opinion just to be safe.

File: depends/packages/openssl.mk (hunk 1 above)
Comment (josephbisch) [inline depends/packages/openssl.mk]: I was trying to fix something that wasn't working by bringing the openssl version in line with that of BC Core and accidentally didn't undo that change. I'll try going back to 1.0.1m or whatever is newest and it should work.

File: depends/patches/python3/0010-cross-darwin-feature.patch (hunk 2 above)
Comment (josephbisch) [inline depends/patches/python3/0010-cross-darwin-feature.patch]: It came from [this Python bug thread](http://bugs.python.org/issue16291).

**Comment by josephbisch** [comment]
What I ended up doing about the PyQt issue is that I skipped the PyQt check (because I realized we don't need to check for the PyQt Python module, because it isn't used as part of the build process, just on the host machine). Then I switched the pyrcc5 check to search on the PATH instead of running some Python code that uses the PyQt module to find the path to pyrcc5.

I think the way it is now makes more sense. Just the stuff used during the build process is checked. I think I'll still check in the PyQt mailing list to be sure there isn't a better way or more standard way of checking for pyrcc5.

File: depends/patches/python3/0010-cross-darwin-feature.patch (hunk 2 above)
Comment (droark) [inline depends/patches/python3/0010-cross-darwin-feature.patch]: Okay. Just looked it over. I assume it works! Nothing jumped out at me as looking blatantly wrong.

File: depends/packages/openssl.mk (hunk 1 above)
Comment (droark) [inline depends/packages/openssl.mk]: Okay. Do you want me to accept this as-is? I'm not worried either way. I just want to be sure I understand your intentions.

Comment (josephbisch) [inline depends/packages/openssl.mk]: I guess you can accept it since I know it works the way it is now and I
think BC Core still uses 1.0.1k.

**Pull request merged**